content/en/blog/_posts/2023-08-21-job-update-post.md

```diff
@@ -0,0 +1,207 @@
+Pods might produce undesired scale ups.
+
+### How can I use it
+
+This is an alpha feature, which you can enable by turning on `JobPodReplacementPolicy`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) in
+your cluster.
+
+Once the feature is enabled you can use it by creating a new Job, which specifies
+`podReplacementPolicy` field as shown here:
+
+```yaml
+kind: Job
+metadata:
+  name: new
+  ...
+spec:
+  podReplacementPolicy: Failed
+  ...
+```
+
+Additionally, you can inspect the `.status.terminating` field of a Job. The value
+of the field is the number of Pods owned by the Job that are currently terminating.
+
+```shell
+kubectl get jobs/myjob -o yaml
+```
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+status:
+  terminating: 3 # three Pods are terminating and have not yet reached the Failed phase
+```
+
+This can be particularly useful for external queueing controllers, such as
+[Kueue](https://github.com/kubernetes-sigs/kueue), that would calculate the
+quota and suspend the start of a new Job until the resources are reclaimed from
+the currently terminating Job.
+
+### How can I learn more?
+
+- Read the KEP: [PodReplacementPolicy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated)
+
+## Job Backoff Limit per Index
+
+### What problem does this solve?
+
+By default, pod failures for [Indexed Jobs](/docs/concepts/workloads/controllers/job/#completion-mode)
+are counted towards the global limit of retries, represented by `.spec.backoffLimit`.
+This means, that if there is a consistently failing index, it is restarted
+repeatedly until it exhausts the limit. Once the limit is exceeded the entire
+Job is marked failed and some indexes may never be even started.
+
+This is problematic for use cases where you want to handle pod failures for
+every index independently. For example, if you use Indexed Jobs for running
+integration tests where each index corresponds to a testing suite. In that case,
+you may want to account for possible flake tests allowing for 1 or 2 retries per
+suite. Additionally, there might be some buggy suites, making the corresponding
+indexes fail consistently. In that case you may prefer to terminate retries for
+that indexes, yet allowing other suites to complete.
+
+The feature allows you to:
+* complete execution of all indexes, despite some indexes failing,
+* better utilize the computational resources by avoiding unnecessary retries of consistently failing indexes.
+
+### How to use it?
+
+This is an alpha feature, which you can enable by turning on the
+`JobBackoffLimitPerIndex`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+in your cluster.
+
+Once the feature is enabled, you can create an Indexed Job with the
+`.spec.backoffLimitPerIndex` field specified.
+
+#### Example
+
+The following example demonstrates how to use this feature to make sure the
```

Comment:
```suggestion
The following example demonstrates how to use this feature to make it more likely that a
```

Reply:
Yeah, to some extent it is still possible that not all indexes would execute if the job fails for another reason that hitting the limit. For example, due to timeout, or is manually deleted by a user.

Reply:
But I don't really like the formulation, "more likely" because it feels like this was flaky :). It is guaranteed now, provided there is no other independent reason for job failure, than pod failures themselves.

Reply:
Added the clarification in parentheses, PTAL

Reply:
This example doesn't execute all indexes - it **fails**. The text “make sure” implies otherwise.